src/Framework/IBuildEngine6.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    /// This interface extends <see cref="IBuildEngine5" /> to allow tasks to get the current project's global properties.
+    /// </summary>
+    public interface IBuildEngine6 : IBuildEngine5
+    {
+        /// <summary>
+        /// Gets the global properties for the current project.
+        /// </summary>
+        /// <returns>A <see cref="Dictionary{String, String}" /> containing the global properties of the current project.</returns>
+        Dictionary<string, string> GetGlobalProperties();
```

Comment:
```suggestion
        ReadOnlyDictionary<string, string> GetGlobalProperties();
```

?

Reply:
ReadOnlyDictionary isn't in .NET 3.5 so the MSBuildTaskHost wouldn't compile :(

Reply:
So I made sure that I copy the dictionary before returning it

Reply:
Nevermind, you are right!  I was trying to use `IReadOnlyDictionary` in `TaskHostConfiguration` which was causing me build errors in MSBuildTaskHost.  But I can just use a Dictionary in that class and use IReadOnlyDictionary everywhere else.